staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -376,3 +385,22 @@ func (r *Reflector) setLastSyncResourceVersion(v string) {
 		r.metrics.lastResourceVersion.Set(float64(rv))
 	}
 }
+
+func (r *Reflector) incrementAndUpdateConsistentErrors(stopCh <-chan struct{}, err error) {
+	r.updateConsistentErrors(stopCh, err)
+}
+
+func (r *Reflector) resetAndUpdateConsistentErrors(stopCh <-chan struct{}) {
+	// sending a nil error sets the counter to zero
+	r.updateConsistentErrors(stopCh, nil)
+}
+
+func (r *Reflector) updateConsistentErrors(stopCh <-chan struct{}, err error) {
+	select {
+	case <-stopCh:
+		return
+	case r.consistentErrorsCh <- err:
+	default:
+		klog.Warning("TODO: add a meaningful error msg")
```

Comment:
This would happen whenever the channel was full, right?

Reply:
this would happen when the receiver hasn't been started or has died. I'm planing to add `wait.Until(...)` to `run` method to prevent the latter.